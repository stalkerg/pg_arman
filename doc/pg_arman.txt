= pg_arman(1) =

== NAME ==

pg_arman - Backup and recovery manager for PostgreSQL

== SYNOPSIS ==

pg_arman [ OPTIONS ]
    { init |
      backup |
      restore |
      show [ DATE | timeline ] |
      validate [ DATE ] |
      delete DATE }

DATE is the start time of the target backup in ISO-format:
(YYYY-MM-DD HH:MI:SS). Prefix match is used to compare DATE and backup
files.

== DESCRIPTION ==

pg_arman is a utility program to backup and restore PostgreSQL database.

It proposes the following features:

- Backup while database runs including tablespaces with just one
  command
- Recovery from backup with just one command, with customized targets
  to facilitate the use of PITR.
- Support for full and differential backup
- Compression of backup files
- Management of backups with integrated catalog

== COMMANDS ==

pg_arman supports the following commands. See also *OPTIONS* for more
details.

*init*::
    Initialize a backup catalog.

*backup*::
    Take an online backup.

*restore*::
    Perform restore.

*show*::
    Show backup history. The timeline option shows timeline of the backup
    and the parent's timeline for each backup.

*validate*::
    Validate backup files.

*delete*::
    Delete backup files.

=== INITIALIZATION ===

First, you need to create "a backup catalog" to store backup files and
their metadata. It is recommended to setup archive_mode and archive_command
in postgresql.conf before initializing the backup catalog. If the variables
are initialized, pg_arman can adjust the config file to the setting. In this
case, you have to specify the database cluster path for PostgreSQL. Please
specify it in PGDATA environmental variable or -D/--pgdata option.

	$ pg_arman init -B /path/to/backup/

=== BACKUP ===

Backup target can be one of the following types:

- Full backup, backup a whole database cluster.
- Differential backup, backup only files or pages modified after the last
verified backup.

It is recommended to verify backup files as soon as possible after backup.
Unverified backup cannot be used in restore and in differential backup.

=== RESTORE ===

PostgreSQL server should be stopped before performing a restore. If database
cluster still exists, restore command will save unarchived transaction log
and delete all database files. You can retry recovery until a new backup is
taken. After restoring files, pg_arman creates recovery.conf in $PGDATA. The
conf file contains parameters for recovery. It is as well possible to modify
the file manually.

It is recommended to take a full backup as soon as possible after recovery
has succeeded.

If "--recovery-target-timeline" is not specifed, the last checkpoint's
TimeLineID in control file ($PGDATA/global/pg_control) will be the restore
target. If pg_control is not present, TimeLineID in the full backup used by
the restore will be a restore target.


== EXAMPLES ==

To reduce the number of command line arguments, you can set BACKUP_PATH,
an environment variable, to the absolute path of the backup catalog and
write default configuration into ${BACKUP_PATH}/pg_arman.ini.

	$ cat $BACKUP_PATH/pg_arman.ini
	ARCLOG_PATH = /home/postgres/arclog
	BACKUP_MODE = FULL
	COMPRESS_DATA = YES
	KEEP_DATA_GENERATIONS = 3
	KEEP_DATA_DAYS = 120

=== TAKE A BACKUP ===

This example takes a full backup of the whole database. Then, it validates
all unvalidated backups.

	$ pg_arman backup --backup-mode=full
	$ pg_arman validate

=== RESTORE FROM A BACKUP ===

Here are some commands to restore from a backup:

	$ pg_ctl stop -m immediate
	$ pg_arman restore
	$ pg_ctl start

=== SHOW A BACKUP ===

	$ pg_arman show
	===================================================================================
	Start                Mode  Current TLI  Parent TLI  Time    Data   Backup   Status 
	===================================================================================
	2013-12-25 03:02:31  PAGE            1           0    0m   203kB     67MB   DONE
	2013-12-25 03:02:31  PAGE            1           0    0m      0B       0B   ERROR
	2013-12-25 03:02:25  FULL            1           0    0m    33MB    364MB   OK

The fields are:

* Start: start time of backup
* Mode: Mode of backup: FULL (full) or PAGE (page differential)
* Current TLI: current timeline of backup
* Parent TLI: parent timeline of backup
* Time: total time necessary to take this backup
* Data: size of data files
* Log: size of read server log files
* Backup: size of backup (= written size)
* Status: status of backup. Possible values are:
- OK : backup is done and validated.
- DONE : backup is done, but not validated yet.
- RUNNING : backup is running
- DELETING : backup is being deleted.
- DELETED : backup has been deleted.
- ERROR : backup is unavailable because some errors occur during backup.
- CORRUPT : backup is unavailable because it is broken.

When a date is specified, more details about a backup is retrieved:

	$ pg_arman show '2011-11-27 19:15:45'
	# configuration
	BACKUP_MODE=FULL
	COMPRESS_DATA=false
	# result
	TIMELINEID=1
	START_LSN=0/08000020
	STOP_LSN=0/080000a0
	START_TIME='2011-11-27 19:15:45'
	END_TIME='2011-11-27 19:19:02'
	RECOVERY_XID=1759
	RECOVERY_TIME='2011-11-27 19:15:53'
	DATA_BYTES=25420184
	BLOCK_SIZE=8192
	XLOG_BLOCK_SIZE=8192
	STATUS=OK

You can check the "RECOVERY_XID" and "RECOVERY_TIME" which are used for
restore option "--recovery-target-xid", "--recovery-target-time".

The delete command deletes backup files not required by recovery after
the specified date.

== OPTIONS ==

pg_arman accepts the following command line parameters. Some of them can
be also specified as environment variables. See also *PARAMETERS* for the
details.

=== COMMON OPTIONS ===
As a general rule, paths for data location need to be specified as
absolute paths; relative paths are not allowed.

*-D* _PATH_ / *--pgdata*=_PATH_::
    The absolute path of database cluster. Required on backup and
    restore.

*-A* _PATH_ / *--arclog-path*=_PATH_::
    The absolute path of archive WAL directory. Required for restore
    and show command.

*-B* _PATH_ / *--backup-path*=_PATH_::
    The absolute path of backup catalog. This option is mandatory.

*-c* / *--check*::
    If specifed, pg_arman doesn't perform actual jobs but only checks
    parameters and required resources. The option is typically used with
    --verbose option to verify the operation.

*-v* / *--verbose*::
    If specified, pg_arman works in verbose mode.

=== BACKUP OPTIONS ===

*-b* _BACKUPMODE_ / *--backup-mode*=_BACKUPMODE_::
    Specify backup target files. Available options are: "full",
    "page".

*-Z* / *--compress-data*::
    Compress backup files with zlib if specified.

*-C* / *--smooth-checkpoint*::
    Checkpoint is performed on every backups. If the option is specified,
    do smooth checkpoint then. See also the second argument for
    pg_start_backup().

*--validate*::
    Validate a backup just after taking it. Other backups taken
    previously are ignored.

*--keep-data-generations*=_NUMBER_ / *--keep-data-days*=_DAYS_::
    Specify how long backuped data files will be kept.
    --keep-data-generations means number of backup generations.
    --keep-data-days means days to be kept.
    Only files exceeded one of those settings are deleted.

=== RESTORE OPTIONS ===

The parameters whose name start are started with --recovery refer to
the same parameters as the ones in recovery.confin recovery.conf.

*--recovery-target-timeline*=_TIMELINE_::
    Specifies recovering into a particular timeline. If not specified,
    the current timeline is used.

*--recovery-target-time*=_TIMESTAMP_::
    This parameter specifies the timestamp up to which recovery will
    proceed.

*--recovery-target-xid*=_XID::
    This parameter specifies the transaction ID up to which recovery
    will proceed.

*--recovery-target-inclusive*::
    Specifies whether server pauses when recovery target is reached.

=== CATALOG OPTIONS ===

*-a* / *--show-all*::
    Show all existing backups, including the deleted ones.

=== CONNECTION OPTIONS ===
Parameters to connect PostgreSQL server.

*-d* _DBNAME_ / *--dbname*=_DBNAME_::
    The database name to execute pg_start_backup() and pg_stop_backup().

*-h* _HOSTNAME_ / *--host*=_HOSTNAME_::
    Specifies the host name of the machine on which the server is running.
    If the value begins with a slash, it is used as the directory for the
    Unix domain socket.

*-p* _PORT_ / *--port*=_PORT_::
    Specifies the TCP port or local Unix domain socket file extension on
    which the server is listening for connections.

*-U* _USERNAME_ / *--username*=_USERNAME_::
    User name to connect as.

*-w* / *--no-password*::
    Never issue a password prompt. If the server requires password
    authentication and a password is not available by other means such as
    a .pgpass file, the connection attempt will fail. This option can be
    useful in batch jobs and scripts where no user is present to enter a
    password.

*-W* / *--password*::
    Force pg_arman to prompt for a password before connecting to a database.
    This option is never essential, since pg_arman will automatically
    prompt for a password if the server demands password authentication.
    However, pg_arman will waste a connection attempt in order to find out
    if the server wants a password. In some cases it is worth typing -W
    to avoid the extra connection attempt.

=== GLOBAL OPTIONS ===

*--help*::
    Print help, then exit.

*-V* / *--version*::
    Print version information, then exit.

*-!* / *--debug*::
    Show debug information.

== PARAMETERS ==

Some of parameters can be specified as command line arguments, environment
variables or in configuration file as follows:

	Short	Long			Env			File
	-h	--host			PGHOST			No
	-p	--port			PGPORT			No
	-d	--dbname		PGDATABASE		No
	-U	--username		PGUSER			No
					PGPASSWORD		No
	-w	--password					No
	-W	--no-password					No
	-D	--pgdata		PGDATA			Yes
	-B	--backup-path		BACKUP_PATH		Yes
	-A	--arclog-path		ARCLOG_PATH		Yes
	-b	--backup-mode		BACKUP_MODE		Yes
	-Z	--compress-data		COMPRESS_DATA		Yes
	-C	--smooth-checkpoint	SMOOTH_CHECKPOINT	Yes
		--validate	        VALIDATE		Yes
		--keep-data-generations	KEEP_DATA_GENERATIONS	Yes
		--keep-data-days	KEEP_DATA_DAYS		Yes
		--recovery-target-timeline RECOVERY_TARGET_TIMELINE Yes
		--recovery-target-xid	RECOVERY_TARGET_XID	Yes
		--recovery-target-time	RECOVERY_TARGET_TIME	Yes
		--recovery-target-inclusive RECOVERY_TARGET_INCLUSIVE Yes

Variable names in configuration file are the same as long names or names
of environment variables. The password can not be specified in command
line and configuration file for security reason.

This utility, like most other PostgreSQL utilities, also uses the
environment variables supported by libpq (see Environment Variables).

== RESTRICTIONS ==

pg_arman has the following restrictions.

- Requires to read database cluster directory and write backup catalog
  directory. It is usually necessary to mount the disk where backup
  catalog is placed with NFS or related from database server.
- Major versions of pg_arman and server should match.
- Block sizes of pg_arman and server should match.
- If there are some unreadable files/directories in data folder of server
  WAL directory or archived WAL directory, the backup or restore will fail
  depending on the backup mode selected.
- Differential backup is not able to take necessary files after a database
  creation, so take a full backup once a new database is created.

== DETAILS ==

=== RECOVERY TO POINT-IN-TIME ===
pg_arman can recover to point-in-time if timeline, transaction ID, or
timestamp is specified in recovery.conf. xlogdump is a contrib module of
PostgreSQL core that allows checking in the content of WAL files and
determine when to recover. This might help.

=== CONFIGURATION FILE ===
Setting parameters in configuration file is done as "name=value". Quotes
are required if the value contains whitespaces. Comments should start with
"#" and are automatically ignored. Whitespaces and tabs are ignored
excluding values.

=== EXIT CODE ===
pg_arman returns exit codes for each error status.

	Code	Name			Description
	0	SUCCESS			Operation succeeded.
	1	HELP			Print help, then exit
	2	ERROR			Generic error
	3	FATAL			Exit because of repeated errors
	4	PANIC			Unknown critical condition
	10	ERROR_SYSTEM		I/O or system error
	11	ERROR_NOMEM		Out of memory
	12	ERROR_ARGS		Invalid input parameters
	13	ERROR_INTERRUPTED	Interrupted by user (Ctrl+C etc.)
	14	ERROR_PG_COMMAND	SQL error
	15	ERROR_PG_CONNECT	Cannot connect to PostgreSQL server
	20	ERROR_ARCHIVE_FAILED    Cannot archive WAL file
	21	ERROR_NO_BACKUP		Backup file not found
	22	ERROR_CORRUPTED		Backup file is broken
	23	ERROR_ALREADY_RUNNING	Cannot start because another pg_arman
					is running
	24	ERROR_PG_INCOMPATIBLE	Version conflicted with server
	25	ERROR_PG_RUNNING	Error due to server running
	26	ERROR_PID_BROKEN	postmaster.pid is broken

== AUTHOR ==
pg_arman is a fork of pg_rman that was originally written by NTT, now developed
and maintained by Michael Paquier.

Please report bug reports at <https://github.com/michaelpq/pg_arman>.
