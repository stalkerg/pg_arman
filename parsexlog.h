#ifndef PARSEXLOG_H
#define PARSEXLOG_H

#include "access/rmgr.h"
#include "access/xlog_internal.h"
#include "access/xlogreader.h"
#include "access/timeline.h"
#include "catalog/pg_control.h"
#include "catalog/storage_xlog.h"
#include "commands/dbcommands_xlog.h"

extern TimeLineHistoryEntry *timeline_history;
extern int timeline_num_entries;

extern void extractPageMap(const char *datadir, XLogRecPtr startpoint,
			   int tliIndex, XLogRecPtr endpoint);
extern void findLastCheckpoint(const char *datadir, XLogRecPtr searchptr,
				   int tliIndex,
				   XLogRecPtr *lastchkptrec, TimeLineID *lastchkpttli,
				   XLogRecPtr *lastchkptredo);
extern XLogRecPtr readOneRecord(const char *datadir, XLogRecPtr ptr,
			  int tliIndex);

#endif