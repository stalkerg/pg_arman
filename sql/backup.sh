#!/bin/bash

#============================================================================
# This is a test script for backup command of pg_arman.
#============================================================================

# Load common rules
. sql/common.sh backup

# Extra parameters exclusive to this test
SCALE=1
DURATION=10
USE_DATA_CHECKSUM=""

# command line option handling for this script
while [ $# -gt 0 ]; do
	case $1 in
		"-d")
			DURATION=`expr $2 + 0`
			if [ $? -ne 0 ]; then
				echo "invalid duration"
				exit 1
			fi
			shift 2
			;;
		"-s")
			SCALE=`expr $2 + 0`
			if [ $? -ne 0 ]; then
				echo "invalid scale"
				exit 1
			fi
			shift 2
			;;
		"--with-checksum")
			USE_DATA_CHECKSUM="--data-checksum"
			shift
			;;
		*)
			shift
			;;
	esac
done

# Check presence of pgbench command and initialize environment
which pgbench > /dev/null 2>&1
ERR_NUM=$?
if [ $ERR_NUM != 0 ]
then
    echo "pgbench is not installed in this environment."
    echo "It is needed in PATH for those regression tests."
    exit 1
fi

function cleanup()
{
    # cleanup environment
    pg_ctl stop -m immediate > /dev/null 2>&1
    rm -fr ${PGDATA_PATH}
    rm -fr ${BACKUP_PATH}
    rm -fr ${ARCLOG_PATH}
    rm -fr ${SRVLOG_PATH}
    rm -fr ${TBLSPC_PATH}
    mkdir -p ${ARCLOG_PATH}
    mkdir -p ${SRVLOG_PATH}
    mkdir -p ${TBLSPC_PATH}
}

function init_database()
{
    rm -rf ${PGDATA_PATH}
    rm -fr ${ARCLOG_PATH}
    rm -fr ${SRVLOG_PATH}
    rm -fr ${TBLSPC_PATH}
    mkdir -p ${ARCLOG_PATH}
    mkdir -p ${SRVLOG_PATH}
    mkdir -p ${TBLSPC_PATH}

    # create new database cluster
    initdb ${USE_DATA_CHECKSUM} --no-locale -D ${PGDATA_PATH} > ${TEST_BASE}/initdb.log 2>&1
    cp ${PGDATA_PATH}/postgresql.conf ${PGDATA_PATH}/postgresql.conf_org
    cat << EOF >> ${PGDATA_PATH}/postgresql.conf
port = ${TEST_PGPORT}
logging_collector = on
wal_level = hot_standby
log_directory = '${SRVLOG_PATH}'
log_filename = 'postgresql-%F_%H%M%S.log'
archive_mode = on
archive_command = 'cp %p ${ARCLOG_PATH}/%f'
max_wal_size = 512MB
EOF

    # start PostgreSQL
    pg_ctl start -D ${PGDATA_PATH} -w -t 300 > ${TEST_BASE}/pg_ctl.log 2>&1
	mkdir -p ${TBLSPC_PATH}/pgbench
	psql --no-psqlrc -p ${TEST_PGPORT} -d postgres > /dev/null 2>&1 << EOF
CREATE TABLESPACE pgbench LOCATION '${TBLSPC_PATH}/pgbench';
CREATE DATABASE pgbench TABLESPACE = pgbench;
EOF

    pgbench -i -s ${SCALE} -p ${TEST_PGPORT} -d pgbench > ${TEST_BASE}/pgbench.log 2>&1

}

function init_catalog()
{
    rm -fr ${BACKUP_PATH}
    pg_arman init -B ${BACKUP_PATH} --quiet
}

cleanup
init_database
init_catalog

echo '###### BACKUP COMMAND TEST-0001 ######'
echo '###### full backup mode ######'
pg_arman backup -B ${BACKUP_PATH} -b full -p ${TEST_PGPORT} -d postgres --quiet;echo $?
pg_arman validate -B ${BACKUP_PATH} --quiet
pg_arman show -B ${BACKUP_PATH} > ${TEST_BASE}/TEST-0001.log 2>&1
grep -c OK ${TEST_BASE}/TEST-0001.log
grep OK ${TEST_BASE}/TEST-0001.log | sed -e 's@[^-]@@g' | wc -c


echo '###### BACKUP COMMAND TEST-0002 ######'
echo '###### page backup mode ######'
pg_arman backup -B ${BACKUP_PATH} -b page -p ${TEST_PGPORT} -d postgres --quiet;echo $?
pg_arman validate -B ${BACKUP_PATH} --quiet
pg_arman show -B ${BACKUP_PATH} > ${TEST_BASE}/TEST-0002.log 2>&1
grep -c OK ${TEST_BASE}/TEST-0002.log
grep OK ${TEST_BASE}/TEST-0002.log | sed -e 's@[^-]@@g' | wc -c

#archive mode not support in pg_arman
#echo '###### BACKUP COMMAND TEST-0003 ######'
#echo '###### archive backup mode ######'
#pg_arman backup -B ${BACKUP_PATH} -b archive -p ${TEST_PGPORT} -d postgres --quiet;echo $?
#pg_arman validate -B ${BACKUP_PATH} --quiet
#pg_arman show -B ${BACKUP_PATH} > ${TEST_BASE}/TEST-0003.log 2>&1
#grep -c OK ${TEST_BASE}/TEST-0003.log
#grep OK ${TEST_BASE}/TEST-0003.log | sed -e 's@[^-]@@g' | wc -c

#we not support server logs
#echo '###### BACKUP COMMAND TEST-0004 ######'
#echo '###### full backup with server log ######'
#init_catalog
#pg_arman backup -B ${BACKUP_PATH} -b full -s -p ${TEST_PGPORT} -d postgres --quiet;echo $?
#pg_arman validate -B ${BACKUP_PATH} --quiet
#pg_arman show -B ${BACKUP_PATH} > ${TEST_BASE}/TEST-0004.log 2>&1
#grep -c OK ${TEST_BASE}/TEST-0004.log
#grep OK ${TEST_BASE}/TEST-0004.log | sed -e 's@[^-]@@g' | wc -c

echo '###### BACKUP COMMAND TEST-0005 ######'
echo '###### full backup with compression ######'
init_catalog
pg_arman backup -B ${BACKUP_PATH} -b full -Z -p ${TEST_PGPORT} -d postgres --quiet;echo $?
pg_arman validate -B ${BACKUP_PATH} --quiet
pg_arman show -B ${BACKUP_PATH} > ${TEST_BASE}/TEST-0005.log 2>&1
grep -c OK ${TEST_BASE}/TEST-0005.log
grep OK ${TEST_BASE}/TEST-0005.log | grep -c true
grep OK ${TEST_BASE}/TEST-0005.log | sed -e 's@[^-]@@g' | wc -c

echo '###### BACKUP COMMAND TEST-0006 ######'
echo '###### full backup with smooth checkpoint ######'
init_catalog
pg_arman backup -B ${BACKUP_PATH} -b full -C -p ${TEST_PGPORT} -d postgres --quiet;echo $?
pg_arman validate -B ${BACKUP_PATH} --quiet
pg_arman show -B ${BACKUP_PATH} > ${TEST_BASE}/TEST-0006.log 2>&1
grep -c OK ${TEST_BASE}/TEST-0006.log
grep OK ${TEST_BASE}/TEST-0006.log | sed -e 's@[^-]@@g' | wc -c


#--full-backup-on-error option not supported in pg_arman
#echo '###### BACKUP COMMAND TEST-0010 ######'
#echo '###### switch backup mode from page to full ######'
#init_catalog
#echo 'page backup without validated full backup'
#pg_arman backup -B ${BACKUP_PATH} -b page -Z -p ${TEST_PGPORT} -d postgres;echo $?
#echo 'page backup in the same situation but with --full-backup-on-error option'
#pg_arman backup -B ${BACKUP_PATH} -b page -F -Z -p ${TEST_PGPORT} -d postgres;echo $?
#pg_arman validate -B ${BACKUP_PATH} --quiet
#pg_arman show -B ${BACKUP_PATH} > ${TEST_BASE}/TEST-0010.log 2>&1
#grep OK ${TEST_BASE}/TEST-0010.log | grep FULL | wc -l
#grep ERROR ${TEST_BASE}/TEST-0010.log | grep PAGE | wc -l

#archive mode not support in pg_arman
#echo '###### BACKUP COMMAND TEST-0011 ######'
#echo '###### switch backup mode from archive to full ######'
#init_catalog
#echo 'archive backup without validated full backup'
#pg_arman backup -B ${BACKUP_PATH} -b archive -s -Z -p ${TEST_PGPORT} -d postgres;echo $?
#sleep 1
#echo 'archive backup in the same situation but with --full-backup-on-error option'
#pg_arman backup -B ${BACKUP_PATH} -b archive -F -s -Z -p ${TEST_PGPORT} -d postgres;echo $?
#pg_arman validate -B ${BACKUP_PATH} --quiet
#pg_arman show -B ${BACKUP_PATH} > ${TEST_BASE}/TEST-0011.log 2>&1
#grep OK ${TEST_BASE}/TEST-0011.log | grep FULL | wc -l
#grep ERROR ${TEST_BASE}/TEST-0011.log | grep ARCH | wc -l

#not support now in pg_arman
#echo '###### BACKUP COMMAND TEST-0012 ######'
#echo '###### failure in backup with different system identifier database ######'
#init_catalog
#pg_ctl stop -m immediate > /dev/null 2>&1
#init_database
#pg_arman backup -B ${BACKUP_PATH} -b full -p ${TEST_PGPORT} -d postgres --quiet;echo $?


# cleanup
## clean up the temporal test data
pg_ctl stop -m immediate > /dev/null 2>&1
rm -fr ${PGDATA_PATH}
rm -fr ${BACKUP_PATH}
rm -fr ${ARCLOG_PATH}
rm -fr ${SRVLOG_PATH}
rm -fr ${TBLSPC_PATH}
