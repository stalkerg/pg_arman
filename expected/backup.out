\! bash sql/backup.sh
###### BACKUP COMMAND TEST-0001 ######
###### full backup mode ######
NOTICE:  pg_stop_backup complete, all required WAL segments have been archived
0
1
3
###### BACKUP COMMAND TEST-0002 ######
###### page backup mode ######
Find timeline index
NOTICE:  pg_stop_backup complete, all required WAL segments have been archived
0
2
6
###### BACKUP COMMAND TEST-0005 ######
###### full backup with compression ######
NOTICE:  pg_stop_backup complete, all required WAL segments have been archived
0
1
0
3
###### BACKUP COMMAND TEST-0006 ######
###### full backup with smooth checkpoint ######
NOTICE:  pg_stop_backup complete, all required WAL segments have been archived
0
1
3
